### PR TITLE
Use make `fetchCors` a pass-through to `fetch` and deprecate

### PR DESCRIPTION
**Pull request by samholmes**
### CHANGELOG

- Changed: Added a fallback to edge-cors-proxy server to `fetch` method on `EdgeIo`.

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204536097865430

### CODE REVIEW

**Comment by samholmes** [comment]
@paullinator @swansontec  Is there a better place for the hard-coded proxy server URL? Also, should we consider `cors1.edge.app` (I chose `cors10` from what @paullinator had spun up as a testing server at one point).

File: src/io/react-native/react-native-worker.ts [hunk 1]
```diff
@@ -17,6 +17,11 @@ import { hideProperties } from '../hidden-properties'
 import { makeNativeBridge } from './native-bridge'
 import { ClientIo, WorkerApi } from './react-native-types'
 
+// Hard-coded CORS proxy server
+const PROXY_SERVER_URL = 'https://cors10.edge.app/api/v1'
```
[the remark this answers is not in the document]
Comment (paullinator) [reply src/io/react-native/react-native-worker.ts:20]: Also remove /api/v1

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/io/react-native/react-native-worker.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (samholmes) [reply src/io/react-native/react-native-worker.ts]: There is no asyncWaterfall util in edge-core-js, so maybe a simple randomized selection is enough for now.

File: src/io/react-native/react-native-worker.ts [hunk 2]
```diff
@@ -223,7 +223,12 @@ const fetchCorsProxy = async (
   uri: string,
   opts: EdgeFetchOptions = {}
 ): Promise<Response> => {
-  return await window.fetch(PROXY_SERVER_URL, {
+  const randomIndex = Math.min(
+    PROXY_SERVER_URLS.length - 1,
+    Math.floor(Math.random() * PROXY_SERVER_URLS.length)
+  )
+  const proxyServerUrl = PROXY_SERVER_URLS[randomIndex]
+  return await window.fetch(proxyServerUrl, {
```
Comment (paullinator) [inline src/io/react-native/react-native-worker.ts:231]: This doesn't attempt `asyncWaterfall`

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/io/react-native/react-native-worker.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (paullinator) [reply src/io/react-native/react-native-worker.ts]: It's a simple copy/paste from the gui. I think it's worth adding in-case a server goes down.

File: src/io/react-native/react-native-worker.ts [hunk 3]
```diff
@@ -223,16 +225,15 @@ const fetchCorsProxy = async (
...
-    PROXY_SERVER_URLS.length - 1,
-    Math.floor(Math.random() * PROXY_SERVER_URLS.length)
+  const shuffledUrls = shuffle([...PROXY_SERVER_URLS])
+  const tasks = shuffledUrls.map(proxyServerUrl => async () =>
+    await window.fetch(proxyServerUrl, {
+      headers: {
+        ...opts.headers,
+        'x-proxy-url': uri
+      },
+      ...opts
```
Comment (paullinator) [inline src/io/react-native/react-native-worker.ts:235]: oops. Didn't catch this in the original review, but `...opts` here would overwrite `headers` if `opts.headers` existed

```
> opts = { headers: {h1: 1, h2: 2}, opts2: 'opt2' }
{ headers: { h1: 1, h2: 2 }, opts2: 'opt2' }
> fetchOpts = { headers: {...opts.headers}}
{ headers: { h1: 1, h2: 2 } }
> 
> fetchOpts = { headers: {...opts.headers, proxy: 'myuri' }}
{ headers: { h1: 1, h2: 2, proxy: 'myuri' } }
> 
> fetchOpts = { headers: {...opts.headers, proxy: 'myuri' }, ...opts}
{ headers: { h1: 1, h2: 2 }, opts2: 'opt2' }
> 
```

Should put the `...opts` before the `headers`

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/io/react-native/react-native-worker.ts (hunk 3 above)
Comment (samholmes) [reply src/io/react-native/react-native-worker.ts:235]: That was a dumb mistake on my part. Good catch.

**Review by paullinator — APPROVED** [review]

**Pull request merged**